### DISCUSSION TITLE
QoS to ensure all samples are received by data reader, ignoring startup order

### DISCUSSION THREAD

**Discussion by tmayoff** [OpenDDS/OpenDDS#3659] [Q&A] [answered]
For whatever reason setting up QoS policies properly have been eluding me. I'd like for a writer to send samples when it starts, then have the subscriber whenever it starts receive them, with no guarantee of the startup order.

On the writer I have the following QoS:
```
reliability.kind = DDS::ReliabilityQosPolicyKind::RELIABLE_RELIABILITY_QOS;
history.kind = DDS::HistoryQosPolicyKind::KEEP_ALL_HISTORY_QOS;
durability.kind = DDS::DurabilityQosPolicyKind::TRANSIENT_LOCAL_DURABILITY_QOS;
```

I tried also setting the durability the same on the reader, but it just keeps failing with an invalid_qos message (I'm still working with v3.20)

What is the correct QoS policies for this behaviour and would this work with shared memory as the transport as well?

**Comment by mitza-oci** [comment, marked as answer]
There are tests like LargeSample that use `TRANSIENT_LOCAL_DURABILITY_QOS` so that may be a good starting point.  It does need to be set on the reader (each reader can decide to opt in).

**Reply by tmayoff** [reply]
Turns out it was the liveliness QoS not being set properly on the reader, as I'm still on 3.20 I don't have the more detailed logging for that kind of stuff. Either way it works thanks
